packages/nx/src/utils/package-manager.ts

```diff
@@ -478,3 +481,76 @@ export async function packageRegistryPack(
   const tarballPath = stdout.trim();
   return { tarballPath };
 }
+
+/**
+ * This function returns the workspaces defined in the package manager configuration.
+ * @returns workspaces defined in the package manager configuration, null if not found
+ */
+export function getPackageWorkspaces(
+  packageManager: PackageManager = detectPackageManager(),
+  root: string = workspaceRoot
+): string[] | null {
+  let workspaces: string[] | null = null;
+
+  if (
+    packageManager === 'npm' ||
+    packageManager === 'yarn' ||
+    packageManager === 'bun'
+  ) {
+    const packageJson = readPackageJson();
+    workspaces = packageJson.workspaces;
+  } else if (packageManager === 'pnpm') {
+    const yamlPath = join(root, 'pnpm-workspace.yaml');
+    if (existsSync(yamlPath)) {
+      const yamlContent = readFileSync(yamlPath, 'utf-8');
+
+      const yaml = yamlLoad(yamlContent);
+      workspaces = yaml.packages;
+    }
+  }
+
+  return workspaces;
+}
+
+/**
+ * This function adds a package to the workspaces defined in the package manager configuration.
+ * If the package is already included in the workspaces, it will not be added again.
+ * @param packageManager The package manager to use. If not provided, it will be detected based on the lock file.
+ * @param root The directory the commands will be ran inside of. Defaults to the current workspace's root.
+ * @param packagePath The path of the package to add to the workspaces
+ */
+export function addPackagePathToWorkspaces(
+  packageManager: PackageManager = detectPackageManager(),
+  root: string = workspaceRoot,
+  packagePath: string
+): void {
+  let workspaces: string[] = getPackageWorkspaces(packageManager, root) ?? [];
+  const isPkgIncluded = workspaces.some((w) => minimatch(packagePath, w));
+  if (isPkgIncluded) {
+    return;
+  }
+
+  workspaces.push(packagePath);
+  if (
+    packageManager === 'npm' ||
+    packageManager === 'yarn' ||
+    packageManager === 'bun'
+  ) {
+    const packageJson = readPackageJson();
+    const updatedPackageJson = {
+      ...packageJson,
+      workspaces,
+    };
+    const packageJsonPath = join(root, 'package.json');
+    writeJsonFile(packageJsonPath, updatedPackageJson);
+  } else if (packageManager === 'pnpm') {
+    const yamlPath = join(root, 'pnpm-workspace.yaml');
+    let yaml: any = {};
+    if (existsSync(yamlPath)) {
```

Comment:
Would this preserve YAML comments? I haven't tried it yet.

Reply:
changed the code, it now preserves comments